src/helpers/credential.rs

```diff
@@ -269,6 +260,142 @@ impl<'a> Credential<'a> {
         Ok(())
     }
 
+    pub async fn create_self_issued_credential(
+        &mut self,
+        schema_did: &str,
+        credential_subject_str: &str,
+        bbs_secret: &str,
+        bbs_private_key: &str,
+        credential_revocation_did: Option<&str>,
+        credential_revocation_id: Option<&str>,
+        exp_date: Option<&str>,
+    ) -> Result<String, CredentialError> {
+        let credential_subject: CredentialSubject = serde_json::from_str(credential_subject_str)?;
+        let subject_did = credential_subject.id.unwrap_or("did:evan:default".to_string());
+        let subject_did_str = subject_did.as_str();
+        let schema: CredentialSchema = self.get_did_document(schema_did).await?;
+        let issuer_public_key = self
+            .get_issuer_public_key(&subject_did, "#bbs-key-1")
+            .await?;
+        let use_valid_until= if exp_date.is_some() { true } else { false };
+        let offer = self
+            .create_credential_offer(
+                schema_did,
+                use_valid_until,
+                subject_did_str,
+                Option::from(subject_did_str),
+            )
+            .await?;
+        let credential_values_str = serde_json::to_string(&credential_subject.data)?;
+        let request_str = self
+            .create_credential_request(
+                issuer_public_key.as_str(),
+                bbs_secret,
+                &credential_values_str,
+                offer.as_str(),
+                schema_did,
+            ).await?;
+        let request: BbsCredentialRequest = serde_json::from_str(request_str.as_str())?;
+
+        /*
+        /// * `issuer_did` - DID of the issuer
```

Comment:
unused and commented code should be cleaned :)

Reply:
Done, thank you